tests/server.js

```diff
@@ -1,63 +1,87 @@
 import test from 'ava';
 import request from 'supertest';
+import bcrypt from 'bcrypt-nodejs';
+
 import punchcard from '../';
+import database from '../lib/database';
 
-// cue up our application for all tests
-test.beforeEach((t) => {
-  return punchcard().then(app => {
-    // Ignoring this param reassign as it's the standard for AVA to allow users to assign context to the tests
-    t.context.request = request(app); // eslint-disable-line no-param-reassign
 
-    return app;
+const accounts = [
+  {
+    email: 'admin@test.com',
+    password: 'pa55word',
+    role: 'admin',
+  },
+  {
+    email: 'creator@test.com',
+    password: 'pa55word',
+    role: 'creator',
+  },
+];
+
+const userCheck = (usr) => {
+  database.select('*').from('users').where({
+    email: usr.email,
+  })
+  .then(user => {
+    if (user.length < 1) {
+      database('users').insert({
+        email: usr.email,
+        password: bcrypt.hashSync(usr.password),
+        role: usr.role,
+      })
+      .then(() => {
+        console.log('user created?');
+      });
+    }
   });
-});
+};
 
-test.cb('CMS Landing Page', t => {
-  t.context.request
-    .get('/')
-    .end((err, res) => {
-      t.is(err, null, 'Should not have an error');
-      t.is(res.status, 200, 'should return status 200');
-      t.regex(res.text, /DOCTYPE html/, 'should have an html doctype');
-      t.end();
+// cue up our application for all tests
+test.cb.beforeEach((t) => {
+  t.context.agent = request.agent(); // eslint-disable-line
```

Comment:
Appears from [supertest](https://github.com/visionmedia/supertest) that if you're going to use `agent`, needs to be initialized as `request.agent(app);` and have that be used instead of straight `request`